docs/DeveloperGuide.md

```diff
@@ -394,23 +394,48 @@ _{Explain here how the data archiving feature will be implemented}_
 
 #### Implementation
 
-This feature is mainly implemented by the `CustomiseCommand` and `PersonCard` classes. The following methods are the main ones:
+This feature is mainly implemented by the `CustomiseOrderCommand` and `PersonCard` classes. The following methods are the main ones:
 
-* `CustomiseCommand#changeAttributeOrder(Model model)` - saves the new attribute order in the `preferences.json` file.
+* `CustomiseOrderCommand#execute()` - saves the new attribute order in the `preferences.json` file by calling `Model#setGuiSettings()`.
 * `PersonCard#setAttributes()` - reads the order from the `preferences.json` file and builds the `PersonCard` in the order specified.
 
-_{diagrams and more in depth explanation to be added}_
+Given below is an example usage scenario and how this feature behaves at each part of the mechanism.
+
+Step 1. The user launches the application. Under each contacts' name, the current order the user sees is: Tags, phone number, email and address (This is also the default order). The user wants to have each email right below each name instead.
+
+Step 2. The user executes `customise order e/` to make the emails appear right below the name.
+
+* Step 2a. `SoConnectParaser` processes the input and calls the parser `CustomiseCommandParser` to parse `" order e/"`
+* Step 2b. `CustomiseCommandParser` processes the input and calls the parser `CustomiseOrderCommandParser` to parse `" e/"`.
+* Step 2c. `CustomiseOrderCommandParser` processes the input and adds to a list of attributes with `EMAIL` and then fills in the rest of the attributes in the following default order `TAGS`, `PHONE` and lastly `ADDRESS`.
+* Step 2d. `CustomiseOrderCommandParser` creates a `CustomiseOrderCommand` with the list of attributes.
+* Step 2e. `CustomiseOrderCommandParser` returns the `CustomiseOrderCommand`.
+* Step 2f. `Logic` executes the `CustomiseOrderCommand`.
+* Step 2g. `CustomiseOrderCommand#execute()` calls `Model#setGuiSettings()` to save the new attribute order in `preferences.json`.
+
+The following sequence diagram illustrates Steps 2a to 2g:
+
+![CustomiseOrderSequenceDiagram](images/CustomiseOrderSequenceDiagram.png)
+
+**Note:** 
+
+* The lifeline for `CustomiseOrderCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+Step 3. The user sees the new ordering after `PersonCard#setAttributes()` sets the attributes based of the new order in `preferences.json`.
+
+Step 4. The user closes the application.
```

Comment:
Step 4 and 5 should be removed since it implies that the user will always close the application after every reordering. Maybe add the information as a side note instead? Like the explanation about how the lifeline should end at the destroy marker

Reply:
Oh I see, I shall move it under `Note:`.